src/quantuminspire/qiskit/qi_job.py

```diff
@@ -14,8 +14,10 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+from __future__ import annotations
+
 import time
-from typing import List, Optional, Any, Dict, Callable
+from typing import List, Optional, Any, Dict, Callable, TYPE_CHECKING
 
 from qiskit.providers import JobError, JobTimeoutError, JobV1 as Job
 from qiskit.providers.backend import Backend
```

Comment:
This one is not used anymore? Can be deleted

Reply:
Great catch! Done.